tool/clean/clean_iam_roles/clean_iam_roles.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+
+	"github.com/aws/amazon-cloudwatch-agent/tool/clean"
+)
+
+var (
+	roleNamePrefix = "cwagent-"
+)
+
+func main() {
+	log.Print("Begin to clean IAM Roles")
+	expirationDate := time.Now().UTC().Add(clean.KeepDurationSixtyDay)
+	ctx := context.Background()
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Fatalf("unable to load AWS config: %v", err)
+	}
+	client := iam.NewFromConfig(cfg)
+	if err = deleteRoles(ctx, client, expirationDate); err != nil {
+		log.Fatalf("errors cleaning: %v", err)
+	}
+}
+
+func deleteRoles(ctx context.Context, client *iam.Client, expirationDate time.Time) error {
+	var errs error
+	var marker *string
+	for {
+		output, err := client.ListRoles(ctx, &iam.ListRolesInput{Marker: marker})
+		if err != nil {
+			return err
+		}
+		for _, role := range output.Roles {
+			if strings.HasPrefix(*role.RoleName, roleNamePrefix) && expirationDate.After(*role.CreateDate) {
```

Comment:
Can we add some unit tests just to make sure we are deleting the roles older than 60 days?